core/src/main/java/com/linecorp/armeria/client/Http1ResponseDecoder.java

```diff
@@ -197,7 +199,13 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
                         res.startResponse();
                         final ResponseHeaders responseHeaders = ArmeriaHttpUtil.toArmeria(nettyRes);
                         final boolean written;
-                        if (responseHeaders.status().codeClass() == HttpStatusClass.INFORMATIONAL) {
+                        final HttpStatus status = responseHeaders.status();
+
+                        final boolean hasInvalid100ContinueResponse = !handle100Continue(nettyRes, status);
+                        if (hasInvalid100ContinueResponse) {
+                            state = State.DISCARD;
```

Comment:
I believe we shouldn't discard the connection, as the server might provide an immediate response, which is completely valid.
```
Upon receiving an HTTP/1.1 (or later) request that has a method, target URI, and complete header section that contains a 100-continue expectation and an indication that request content will follow, an origin server MUST send either:
- an immediate response with a final status code, if that status can be determined by examining just the method, target URI, and header fields, or
```
https://datatracker.ietf.org/doc/html/rfc9110#section-10.1.1-15.1

Based on the server's response, the following scenarios apply:
- 100 status: Continue to send the body
- 417 status: Send the original request again without the expect header
- Others status cods: Discard the request body and set the `state = State.NEED_HEADERS;` to reuse the connection.

What are your thoughts on this approach?

Reply:
Thank you for your comments!
I didn’t check the details…
Let me confirm and modify them~

Reply:
Thanks! I've revised my comment, so PTAL again. 😄